feat(segmented-button): added large/small

agliga opened the pull request:
<!-- Insert GitHub issue number below -->
Fixes #2064

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Added large size. 

## Notes
* In order to have the sizes be correct based on designs, regular buttons are using 32px (small) button size and then with 4px padding top+bottom create a 40px container. The same with large, 40px (regular) button size with 4px padding top+bottom create a 48px container. I added a comment to address this. 

## Screenshots
<img width="930" alt="Screen Shot 2023-05-25 at 10 52 35 AM" src="https://github.com/eBay/skin/assets/1755269/b7004d17-953d-4710-bcf2-bfc307a739fb">

## Checklist
- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue
- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

LuLaValva reviewed (commented):
Looks good, just a few minor comments

ArtBlue commented on docs/_includes/segmented-buttons.html:
```diff
@@ -32,6 +32,32 @@ <h3 id="button-default">Default Segmented buttons</h3>
+            <div class="segmented-buttons segmented-buttons--large">
+                <ul>
+                    <li><button type="button" class="segmented-buttons__button"
+                            aria-current="true">Day</button></li>
```
I believe `aria-current` is more appropriate for static implementations, like breadcrumbs. @ianmcburnie and/or @saiponnada can verify this, but in this case, should we not opt to use `aria-selected`?

agliga replied:
We should address that in another pr/issue. This isn't really relevant since this deals with the size changes. (No scope creep)
But we should get clarity on this offline and file an issue to fix it if so.

ArtBlue reviewed (commented):
Looks good; just an `a11y` thingy...

agliga commented:
> Looks good, just a few minor comments

Good catch. I addressed them (and added a variable for border radius). I used large variable for that in all cases because the border radius is always 24px on segmented buttons.

LuLaValva reviewed (approved):
Nice changes, thanks Andrew

ianmcburnie commented:
I'd like us to get into the habit of adding introductory text after each heading. As mentioned in another PR, one tactic I like to use is to start with the baseline, then mention the thing that changed or was added since the prior example.

For example:

**Default Segmented Button:** The default segmented button has a 32px height that matches other buttons.
**Segmented Buttons - large:** Use the segmented-buttons--large modifier to create large segemented buttons.
**Segmented buttons - with icons:** Any 24x24 icon (link out to icons) can be added inside of a segmented-buttons__button-cell block.

<img width="1325" alt="Screen Shot 2023-05-26 at 2 59 43 PM" src="https://github.com/eBay/skin/assets/38065/71f6cd39-3fa9-40ff-a25a-5fbaa34806ce">

agliga commented:
> I'd like us to get into the habit of adding introductory text after each heading. As mentioned in another PR, one tactic I like to use is to start with the baseline, then mention the thing that changed or was added since the prior example.
> 
> For example:
> 
> **Default Segmented Button:** The default segmented button has a 32px height that matches other buttons. **Segmented Buttons - large:** Use the segmented-buttons--large modifier to create large segemented buttons. **Segmented buttons - with icons:** Any 24x24 icon (link out to icons) can be added inside of a segmented-buttons__button-cell block.
> 
> <img alt="Screen Shot 2023-05-26 at 2 59 43 PM" width="1325" src="https://user-images.githubusercontent.com/38065/241332288-71f6cd39-3fa9-40ff-a25a-5fbaa34806ce.png">

Done

agliga commented:
FYI, the regular one is 40px height (the button part is 32px but the overall segmented button with padding turns out to 40), and the large one is 48px (button part is 40px)

ianmcburnie reviewed (approved):
Missed on piece of paragraph text (for segmented buttons in fixed width container), but looks great otherwise!
